### PR TITLE
feature: dependabot integ - move all deps to requirements.txt

### PR DESCRIPTION
**Pull request by mufaddal-rohawala**
*Issue #, if available:*
https://issues.amazon.com/issues/V534089506

*Description of changes:*
Dependabot currently [does not track](https://github.com/aws/sagemaker-python-sdk/network/dependencies) extras and tox.ini dependencies. This PR helps track the dependencies for security alerts. The changes are backwards compatible.

*Testing done:*
Ran unit tests locally. Also checked manual runs for: `pip install -e .'[local]' `, etc.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ahsan-z-khan — APPROVED** [review]

**Review by jeniyat — APPROVED** [review]

**Review by navinsoni — CHANGES_REQUESTED** [review]
is there a specific reason we have coverage in multiple files
- requirements/tox/clean_requirements.txt
- requirements/tox/coverage_requirements.txt

File: setup.py [hunk 1]
```diff
@@ -47,43 +61,15 @@ def read_version():
...
 extras = {
-    "local": [
-        "urllib3==1.26.8",
-        "docker-compose==1.29.2",
-        "docker~=5.0.0",
-        "PyYAML==5.4.1",  # PyYAML version has to match docker-compose requirements
-    ],
-    "scipy": ["scipy==1.5.4"],
+    "local": read_requirements("requirements/extras/local_requirements.txt"),
+    "scipy": read_requirements("requirements/extras/spicy_requirements.txt"),
```
Comment (navinsoni) [inline setup.py:67]: did you mean filename to be spicy :)

File: requirements/tox/mypy_requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+mypi
```
Comment (navinsoni) [inline requirements/tox/mypy_requirements.txt:1]: can you please pin mypi

File: requirements/extras/test_requirements.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+tox==3.24.5
+flake8==4.0.1
+pytest==6.0.2
+pytest-cov==3.0.0
+pytest-rerunfailures==10.2
+pytest-timeout==2.1.0
+pytest-xdist==2.4.0
+coverage>=5.2, <6.2
```
Comment (navinsoni) [inline requirements/extras/test_requirements.txt:8]: mismatch for coverage version here and in other 2 files

Comment (mufaddal-rohawala) [reply requirements/extras/test_requirements.txt:8]: Thanks Will make this consistent.

File: requirements/tox/mypy_requirements.txt (hunk 2 above)
Comment (mufaddal-rohawala) [reply requirements/tox/mypy_requirements.txt:1]: Sure, kept it as is, as it's unused but I agree its a good idea to pin.

File: requirements/tox/spelling_requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+pyenchant
+pylint
```
[the remark this answers is not in the document]
Comment (mufaddal-rohawala) [reply requirements/tox/spelling_requirements.txt:2]: Sure, kept it as is, as it's unused but I agree its a good idea to pin.

File: setup.py (hunk 1 above)
Comment (mufaddal-rohawala) [reply setup.py:67]: ah good catch :-)

**Review by navinsoni — APPROVED** [review]

**Pull request merged**